### PR TITLE
- APEX-263 Fixed case where double checkpointing could occurr.

### PR DESCRIPTION
**Pull request by ilooner**

### CODE REVIEW

**Comment by ilooner** [comment]
This is for review only @PramodSSImmaneni and @243826 Please this is a small fix. I will work on writing a unit test in the mean time.

This fix is working in an application test.

**Comment by ilooner** [comment]
@PramodSSImmaneni @243826 fixed exactly once case and added unit tests. Please review I think it is ready to merge.

File: engine/src/test/java/com/datatorrent/stram/engine/GenericNodeTest.java [hunk 1]
```diff
@@ -296,4 +323,67 @@ public void run()
...
+        ab.set(true);
+        in.activate();
+        in.run();
+        in.deactivate();
+      }
+
+    };
+
+    t.start();
+    Thread.sleep(3000);
```
Comment (pramodin) [inline engine/src/test/java/com/datatorrent/stram/engine/GenericNodeTest.java]: Wait till 3 windows instead of fixed timeout

Comment (ilooner) [inline engine/src/test/java/com/datatorrent/stram/engine/GenericNodeTest.java]: @PramodSSImmaneni made the change

**Pull request merged**